### PR TITLE
Add hook for enterprise services started at launch, remove unused (in…

### PR DESCRIPTION
**Pull request by dadlerj**
… OSS) and incorrect max user count reporting

> This PR does not need to update the CHANGELOG because there are no user-facing changes

### CODE REVIEW

File: cmd/frontend/hooks/hooks.go [hunk 1]
```diff
@@ -5,3 +5,7 @@ import "net/http"
 // PreAuthMiddleware is an HTTP handler middleware that, if set, runs just before auth-related
 // middleware. The client is not yet authenticated when PreAuthMiddleware is called.
 var PreAuthMiddleware func(http.Handler) http.Handler
+
+// EnterpriseLaunch is run as a goroutine on product launch, providing a hook for Enterprise
+// background services.
+var EnterpriseLaunch func()
```
Comment (slimsag) [inline cmd/frontend/hooks/hooks.go]: It seems to me that this should be a list of functions. After all, someone may want to add more than one here in the future.

File: cmd/frontend/internal/cli/serve_cmd.go [hunk 2]
```diff
@@ -162,6 +163,9 @@ func Main() error {
 	goroutine.Go(mailreply.StartWorker)
 	go updatecheck.Start()
 	go useractivity.MigrateUserActivityData(context.Background())
+	if hooks.EnterpriseLaunch != nil {
+		go hooks.EnterpriseLaunch()
```
Comment (slimsag) [inline cmd/frontend/internal/cli/serve_cmd.go]: use `goroutine.Go(hooks.EnterpriseLaunch)` here instead, that way if the goroutine crashes it won't take down the server.

File: cmd/frontend/hooks/hooks.go (hunk 1 above)
Comment (dadlerj) [reply cmd/frontend/hooks/hooks.go]: Why prefer adding that structure here, to just calling out to multiple other functions in the enterprise func? 

E.g., it's not like we created a general purpose list to hold all possible startup/launch commands that gets iterated over in `serve_cmd.go` — we just call out to all of them directly, from that central spot. The same structure is what I'd been envisioning here.

Comment (slimsag) [reply cmd/frontend/hooks/hooks.go]: Oh I see. In that case, I think this should not be ran in a goroutine. The docs here should state that this is for launching other background services in a goroutine. I think this also should not have Enterprise in the name, I would change this to:

```Go
// AfterDBInit is called after the database is initialized, and can be used to
// e.g. launch background services that depend on the database.
var AfterDBInit func()
```

**Comment by dadlerj** [comment]
Now that all code is in this repo, is this PR totally unnecessary?

**Pull request merged**